INSTALL.md

```diff
@@ -0,0 +1,271 @@
+# Installing google-cloud-cpp
+
+By default google-cloud-cpp libraries download and compile all their
+dependencies ([see below](#required-libraries) for a complete list). This makes
+it easier for users to "take the library for a spin", and works well for users
+that "Live at Head", but does not work for package maintainers or users that
+prefer to compile their dependencies once and install them in `/usr/local/` or a
+similar directory.
+
+This document provides instructions to install the dependencies of
+google-cloud-cpp.
+
+## Installing google-cloud-cpp
+
+**If** all the dependencies of `google-cloud-cpp` are installed and provide
+CMake support files, then compiling and installing the libraries
+requires two commands:
+
+```bash
+cmake -H. -Bbuild-output-for-install \
+    -DGOOGLE_CLOUD_CPP_DEPENDENCY_PROVIDER=package
+cmake --build build-output-for-install --target install
+```
+
+Unfortunately getting your system to this state may require multiple steps,
+the following sections describe how to install `google-cloud-cpp` on several
+platforms.
+
+## Table of Contents
+
+- [Installing google-cloud-cpp](#installing-google-cloud-cpp)
+  - [Fedora 29](#fedora-29)
+  - [Ubuntu (Bionic Beaver)](#ubuntu-bionic-beaver)
+  - [CentOS 7](#ubuntu-bionic-beaver)
+
+## Installing google-cloud-cpp
+
+#### Required Libraries
+
+`google-cloud-cpp` directly depends on the following libraries:
+
+| Library | Minimum version | Description |
+| ------- | --------------: | ----------- |
+| gRPC    | 1.17.x | gRPC++ for Cloud Bigtable |
+| libcurl | 7.47.0  | HTTP client library for the Google Cloud Storage client |
+| crc32c  | 1.0.6 | Hardware-accelerated CRC32C implementation |
+| OpenSSL | 1.0.2 | Crypto functions for Google Cloud Storage authentication |
+
+Note that these libraries may have complex dependencies themselves, the
+following instructions include steps to install these dependencies too.
+When possible, the instructions below prefer to use pre-packaged versions of
+these libraries. In some cases the packages do not exist, or the package
+versions are too old. If this is the case, the instructions describe how you
+can manually download and install these packages. 
+
+#### Fedora (29)
+
+Fedora includes packages for gRPC, libcurl, and OpenSSL that are recent enough
+for `google-cloud-cpp`. Install these libraries and the necessary development
+tools:
+
+```bash
+dnf makecache
+dnf install -y automake cmake gcc-c++ git grpc-devel grpc-plugins \
+        libcurl-devel make openssl-devel pkgconfig protobuf-compiler tar wget \
+        which zlib-devel
+```
+
+**crc32c**: there is no Fedora package for this library. To install use:
```

Comment:
**crc32c** has the same size as `Fedora (29)`. Could you change `crc32c` into a heading similar to the following?

### Fedora (29) 

#### crc32c

Reply:
Done.